axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/invoice/payment/InvoicePaymentMoveCreateServiceBankPayImpl.java

```diff
@@ -145,4 +146,17 @@ public void createBankOrder(InvoicePayment invoicePayment)
       bankOrderValidationService.confirm(bankOrder);
     }
   }
+
+  @Override
+  public String getOriginFromInvoicePayment(InvoicePayment invoicePayment) {
+    Optional<String> bankOrderSeq =
+        Optional.of(invoicePayment)
+            .map(InvoicePayment::getBankOrder)
+            .map(BankOrder::getBankOrderSeq);
+    if (bankOrderSeq.isPresent()) {
+      return bankOrderSeq.get();
+    }
+
+    return super.getOriginFromInvoicePayment(invoicePayment);
```

Comment:
can be simplified with `.orElseGet(() -> super.getOriginFromInvoicePayment(invoicePayment))`